### PR TITLE
Add/update documentation for content invalidation jobs in TO API v4

### PR DESCRIPTION
**Pull request by ocket8888**
This PR makes updates to the documentation for Content Invalidation Jobs as they appear in the unstable TO API version 4.0, including the "Quick How-To" guide, the API documentation, adds an overview page and a glossary term, and links usages of the term to the glossary. This also adds a couple missing GoDoc comments, and changes the `InvalidationJobV4.String` behavior to output a representation without any control characters (previously contained a couple dozen tabs and about a half dozen newlines), adding an example/test for the new behavior.

<hr/>

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Control Cache Config (T3C, formerly ORT)
- Traffic Control Health Client (tc-health-client)
- Traffic Control Client 
- Traffic Monitor
- Traffic Ops
- Traffic Portal
- Traffic Router
- Traffic Stats
- Grove
- CDN in a Box
- Automation 
- unknown

## PR submission checklist
- [x] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: docs/source/api/v3/jobs.rst [hunk 1]
```diff
@@ -291,7 +292,7 @@ Response Structure
 	Content-Length: 234
 
 	{ "alerts": [{
-		"text": "Content invalidation job updated",
+		"text": "Content invalidation :term:`Content Invalidation Job` updated",
```
Comment (TaylorCFrey) [inline docs/source/api/v3/jobs.rst:295]: I think this should remain _job_, no `:term:Content Invalidation Job`

File: docs/source/api/v2/jobs.rst [hunk 2]
```diff
@@ -291,7 +292,7 @@ Response Structure
 	Content-Length: 234
 
 	{ "alerts": [{
-		"text": "Content invalidation job updated",
+		"text": "Content invalidation :term:`job` updated",
```
Comment (TaylorCFrey) [inline docs/source/api/v2/jobs.rst:295]: I think this should remain _job_, no `:term:`

File: docs/source/api/v2/jobs.rst [hunk 3]
```diff
@@ -371,7 +372,7 @@ Response Structure
 
 	{ "alerts": [
 		{
-			"text": "Content invalidation job was deleted",
+			"text": "Content invalidation :term:`job` was deleted",
```
Comment (TaylorCFrey) [inline docs/source/api/v2/jobs.rst:375]: I think this should remain _job_, no `:term:`

File: docs/source/api/v3/jobs.rst [hunk 4]
```diff
@@ -371,7 +372,7 @@ Response Structure
 
 	{ "alerts": [
 		{
-			"text": "Content invalidation job was deleted",
+			"text": "Content invalidation :term:`Content Invalidation Job` was deleted",
```
Comment (TaylorCFrey) [inline docs/source/api/v3/jobs.rst:375]: I think this should remain _job_, no `:term:Content Invalidation Job`

File: docs/source/api/v4/jobs.rst [hunk 5]
```diff
@@ -21,38 +21,37 @@
...
+	+----------------------+----------+--------------------------------------------------------------------------------------------------------------------------------------+
+	| maxRevalDurationDays | no       | Return only :term:`Content Invalidation Jobs` with a :ref:`job-start-time` that is within the window defined by the                  |
+	|                      |          | ``maxRevalDurationDays`` :term:`Parameter` in :ref:`the-global-profile`                                                              |
+	+----------------------+----------+--------------------------------------------------------------------------------------------------------------------------------------+
+	| userId               | no       | Return only :term:`Content Invalidation Jobs` created by the user identified by this integral, unique identifier                     |
+	+----------------------+----------+--------------------------------------------------------------------------------------------------------------------------------------+
 
 
 .. code-block:: http
```
Comment (TaylorCFrey) [inline docs/source/api/v4/jobs.rst:57]: Looks like this mighta been missed, but the Response Structure for `GET` has changed here. The fields returned are:

assetUrl (Remains the same)
createdBy (Remains the same)
deliveryService (Remains the same)
~keyword~ (Removed)
~parameters~ (Removed)
startTime (Remains the same)
ttlHours (Added, replaces `parameters`)
invalidationType (Added) -`REFRESH` | `REFETCH`

Also, the response example will need to be updated too.

A crude example of the returned object:
```json
{
      "id": 9,
      "assetUrl": "http://test-ds.local.tld/another/other/to/content\\.jpg",
      "createdBy": "admin",
      "deliveryService": "test-ds",
      "ttlHours": 24,
      "invalidationType": "REFRESH",
      "startTime": "2021-11-05T10:24:00.37-06:00"
  }
  ```

File: docs/source/api/v4/jobs.rst [hunk 6]
```diff
@@ -109,269 +108,236 @@ Response Structure
...
-
-	PURGE
-		This job will prevent caching of URLs matching the ``assetUrl`` until it is removed (or its Time to Live expires)
-
-:parameters: A string containing key/value pairs representing parameters associated with the job - currently only uses Time to Live e.g. ``"TTL:48h"``
-:startTime:  The date and time at which the job began, in a non-standard format
+:assetUrl:         The :ref:`job-asset-url`
+:createdBy:        The :ref:`job-created-by`
+:deliveryService:  The :ref:`job-ds`
+:id:               The :ref:`job-id`.
```
Comment (TaylorCFrey) [inline docs/source/api/v4/jobs.rst:249]: Extra 'full stop' at the end

File: docs/source/api/v4/jobs.rst [hunk 7]
```diff
@@ -109,269 +108,236 @@ Response Structure
...
 	}}
 
 
 ``DELETE``
 ==========
-Deletes a content invalidation job.
+Deletes a :term:`Content Invalidation Job`.
 
-.. tip:: Content invalidation jobs that have passed their :abbr:`TTL (Time To Live)` are not automatically deleted - for record-keeping purposes - so use this to clean up old jobs that are no longer useful.
+.. tip:: Content :term:`Content Invalidation Jobs` that have passed their :abbr:`TTL (Time To Live)` are not automatically deleted - for record-keeping purposes - so use this to clean up old jobs that are no longer useful.
```
Comment (TaylorCFrey) [inline docs/source/api/v4/jobs.rst:288]: Extra `Content` at the beginning.

File: docs/source/overview/profiles_and_parameters.rst [hunk 8]
```diff
@@ -708,6 +707,15 @@ header
 
 	.. caution:: If a single :ref:`Profile <profiles>` has multiple "header" Parameters for the same `Config File`_ with different :ref:`Values <parameter-value>`, the actual header is undefined (but will be one of those Parameters' :ref:`Values <parameter-value>`).
 
+.. _parameter-name-refetch_enabled:
```
Comment (TaylorCFrey) [inline docs/source/overview/profiles_and_parameters.rst:710]: This is really eloquently explained. Nice!

File: lib/go-tc/invalidationjobs.go [hunk 9]
```diff
@@ -511,18 +515,16 @@ type InvalidationJobV40 struct {
...
+// representation of the InvalidationJobV4.
 func (job InvalidationJobV4) String() string {
-
-	ID := strconv.FormatUint(job.ID, 10)
-	TTLHours := strconv.FormatUint(uint64(job.TTLHours), 10)
-	StartTime := job.StartTime.String()
-
-	return fmt.Sprintf("{\nID:\t\t\t\t\t%s,\nAssetURL:\t\t\t%s,\nCreatedBy:\t\t\t%s,\nDeliveryService:\t%s,\nTTLHours:\t\t\t%s,\nInvalidationType:\t%s,\nStartTime:\t\t\t%s,\n}",
-		ID,
+	return fmt.Sprintf(`InvalidationJobV4{ID: %d, AssetURL: "%s", CreatedBy: "%s", DeliveryService: "%s", TTLHours: %d, InvalidationType: "%s", StartTime: "%s"}`,
```
Comment (TaylorCFrey) [inline lib/go-tc/invalidationjobs.go:521]: Nice cleanup.

File: docs/source/overview/jobs.rst [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+.. _jobs:
+
+*************************
+Content Invalidation Jobs
+*************************
+:dfn:`Content Invalidation Jobs`, or simply "jobs" as they are sometimes known, are ways of forcing :term:`cache servers` to treat content as no longer valid, bypassing their normal caching policies.
+
+In general, this *should* be unnecessary, because a well-behaved :term:`Origin` *should* be setting its HTTP caching headers properly, so that content is only considered valid for some appropriate time intervals. Occasionally, however, an :term:`Origin` will be too optimistic with its caching instructions, and when content needs to be updated, :term:`cache servers` need to be informed that they must check back with the :term:`Origin`. Content Invalidation Jobs allow this to be done for specific patterns of assets, so that :term:`cache servers` will check back in with the :term:`Origin` and verify that the content they have cached is still valid.
+
+The model for Content Invalidation Job as API objects is given in :ref:`jobs-model` as a TypeScript interface.
```
Comment (TaylorCFrey) [inline docs/source/overview/jobs.rst:25]: This line is resulting in a duplicated "as a Typescript interface"...

```
[...] given in Content Invalidation Job as a Typescript interface. as a TypeScript interface.
```

File: docs/source/overview/jobs.rst [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+		invalidationType: "REFRESH" | "REFETCH";
+		startTime: Date; // RFC3339 string
+		ttlHours: number;
+	}
+
+.. _job-asset-url:
+
+Asset URL
+---------
+This property only appears in responses from the :ref:`to-api`, and has no effect in requests. The :dfn:`Asset URL` is constructed from the `Regular Expression`_ used in the creation of a Content Invalidation Job and the :ref:`ds-origin-url` of the :term:`Delivery Service` for which it was created. It is a URL that has a valid regular expression as its path (and may not be "percent-encoded" where a normal URL typically would be). Requests from CDN clients for content that matches this pattern will trigger Content Invalidation behavior.
```
Comment (TaylorCFrey) [inline docs/source/overview/jobs.rst:59]: It is possible to edit this field in a `PUT` request.

File: docs/source/overview/jobs.rst [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+.. _job-created-by:
+
+Created By
+----------
+The username of the user who created the Content Invalidation Job is stored as the :dfn:`Created By` property of the Content Invalidation Job.
```
Comment (TaylorCFrey) [inline docs/source/overview/jobs.rst:65]: Not sure if you want to add the `:term:` for both _Content Invalidation Job_ here, or throughout this page really. Given how consistent it is used throughout the API docs and elsewhere I thought I'd bring it up, however I am fine with it either way here.

File: docs/source/overview/jobs.rst [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+
+.. _job-ds:
+
+Delivery Service
+----------------
+A Content Invalidation Job can only act on content for a single :term:`Delivery Service` - invalidating content for multiple :term:`Delivery Services` requires multiple Content Invalidation Jobs. The :dfn:`Delivery Service` property of a Content Invalidation Job holds the :ref:`ds-xmlid` of the :term:`Delivery Service` on which it operates.
```
Comment (TaylorCFrey) [inline docs/source/overview/jobs.rst:71]: Same `:term:` question as above and below. Fine either way.

File: docs/source/overview/jobs.rst [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+:dfn:`Invalidation Type` defines how a :term:`cache server` should go about ensuring that its cache is valid.
+
+The normal operating mode for a Content Invalidation Job is to force the :term:`cache server` to send a request to the :term:`Origin` to verify that its cache is valid. If that is the case, no extra work is done and business as usual resumes. However, some :term:`Origins` are misconfigured and do not respond as required by HTTP specification. In this case, it is strongly advised to fix the :term:`Origin` so that it properly implements HTTP. However, if an :term:`Origin` is sending cache-able responses to requests, and cannot be trusted to verify the validity of cached content based on cache-controlling HTTP headers (e.g. :mailheader:`If-Modified-Since`) instead returning responses like ``304 Not Modified`` *even when the content has in fact been modified*, **and** if correcting this behavior is not an option, then the :term:`cache server` may be forced to pretend that the content it has was actually invalidated by the :term:`Origin` and must be completely re-fetched.
+
+The two values allowed for a Content Invalidation Job's Invalidation Type are:
+
+REFRESH
```
Comment (TaylorCFrey) [inline docs/source/overview/jobs.rst:92]: The explanation is excellent.

**Review by TaylorCFrey — CHANGES_REQUESTED** [review]

File: docs/source/admin/quick_howto/content_invalidation.rst [hunk 15]
```diff
@@ -22,8 +22,6 @@ Invalidating content on the CDN is sometimes necessary when the :term:`Origin` w
 
 .. impl-detail:: Given the size of a typical Traffic Control CDN and the amount of content that can be cached in it, removing the content from all the caches may take a long time. To speed up content invalidation, Traffic Control does not try to remove the content from the caches, but it makes the content inaccessible using the `regex_revalidate plugin for Apache Traffic Server <https://docs.trafficserver.apache.org/en/8.0.x/admin-guide/plugins/regex_revalidate.en.html>`_. This forces a "re-validation" of the content.
 
-.. Warning:: This method forces :term:`cache servers` to "re-validate" content, so in order to work properly the :term:`Origin` needs to support revalidation according to section 4.3.2 of :rfc:`7234`.
-
```
[the remark this answers is not in the document]
Comment (ocket8888) [reply docs/source/admin/quick_howto/content_invalidation.rst:26]: I moved it to the section on "Invalidation Type" in the overview, which is linked to now from this section

File: docs/source/admin/traffic_portal/usingtrafficportal.rst [hunk 16]
```diff
@@ -645,21 +645,21 @@ Tools
...
+Here, specific assets can be invalidated in all caches of a :term:`Delivery Service`, forcing content to be updated from the origin. Specifically, this *doesn't* mean that :term:`cache servers` will immediately remove items from their caches, but rather will fetch new copies whenever a request is made matching the 'Asset URL' regular expression. This behavior persists until the :term:`Content Invalidate Content Job`'s :ref:`job-ttl` expires.
 
-.. Warning:: This method forces :term:`cache servers` to "re-validate" content, so in order to work properly the :term:`Origin` needs to support revalidation according to section 4.3.2 of :rfc:`7234`.
+.. warning:: This method forces :term:`cache servers` to "re-validate" content, so in order to work properly the :term:`Origin` needs to support revalidation according to section 4.3.2 of :rfc:`7234`.
 
 Each entry in the table on this page has the following fields:
 
-:term:`Delivery Service`: The :term:`Delivery Service` to which to apply this Invalidate Content Job
```
[the remark this answers is not in the document]
Comment (ocket8888) [reply docs/source/admin/traffic_portal/usingtrafficportal.rst:654]: I think that was probably my fault, from a search/replace I did back when I added "Delivery Service" to the glossary.

File: docs/source/overview/jobs.rst (hunk 12 above)
Comment (ocket8888) [reply docs/source/overview/jobs.rst:65]: In general I don't use `term` for the overview section that describes the term. The flow of understanding for a term with an overview section is

1. See the term
1. Click on it, which takes you to a one-paragraph (max) glossary entry explanation
1. Click on the seealso link to go to the overview section if you need more detailed information

doesn't make sense to me to put a loop in that flow.

**Review by TaylorCFrey — APPROVED** [review]
LGTM! Changes incorporated, builds successfully.

**Pull request merged**